src/components/SQLBuilderSelectRow.tsx

```diff
@@ -0,0 +1,123 @@
+import { SelectableValue, toOption } from '@grafana/data';
+import { AccessoryButton, EditorField, Stack } from '@grafana/experimental';
+import { Button, Select } from '@grafana/ui';
+import { BigQueryAPI } from 'api';
+import { QueryEditorExpressionType, QueryEditorFunctionExpression } from 'expressions';
+import React from 'react';
+import { useAsync } from 'react-use';
+import { createFunctionField, toRawSql } from 'utils/sql.utils';
+import { BigQueryQueryNG, QueryWithDefaults } from '../types';
+import { BQ_AGGREGATE_FNS } from './query-editor-raw/bigQueryFunctions';
+
+interface SQLBuilderSelectRowProps {
+  query: QueryWithDefaults;
+  apiClient: BigQueryAPI;
+  onQueryChange: (query: BigQueryQueryNG) => void;
+}
+
+export function SQLBuilderSelectRow({ query, apiClient, onQueryChange }: SQLBuilderSelectRowProps) {
+  const state = useAsync(async () => {
+    if (!query.location || !query.dataset || !query.table) {
+      return;
+    }
+    const columns = await apiClient.getColumns(query.location, query.dataset, query.table);
+    return columns.map<SelectableValue<string>>(toOption);
+  }, [apiClient, query.dataset, query.location, query.table]);
```

Comment:
Let's make this a separate `useColumns` hook - exactly the same implementation is used in `SQLOrderByRow` and `SQLGroupByRow`